math/dec.go

```diff
@@ -368,12 +368,25 @@ func (x Dec) Reduce() (Dec, int) {
 
 // Marshal serializes the decimal value into a byte slice in a text format.
 // This method ensures the decimal is represented in a portable and human-readable form.
-func (d Dec) Marshal() ([]byte, error) {
-	return d.dec.MarshalText()
+// The output may be in scientific notation if the number's magnitude is very large or very small.
+// Specifically, scientific notation is used for numbers with an absolute value >= 1e6 or < 1e-5.
```

Comment:
If this is used for token values it might be somewhat non-intuitive as these often at least 6 or 9 digits. What are we optimizing for here? Storage space or human readability?

Reply:
hmm yes good point. Client libraries would have to handle this notation, too when submitted as json. What do you suggest here re format?